corehq/apps/couch_sql_migration/tests/test_migration.py

```diff
@@ -623,3 +623,111 @@ def _validate_ledger_data(self, state_dict, expected):
 
     def _get_ledger_state(self, case_id):
         return LedgerAccessors(self.domain_name).get_case_ledger_state(case_id)
+
+
+class TestLockingQueues(TestCase):
+    def setUp(self):
+        self.queues = PartiallyLockingQueue()
+
+    def _add_to_queues(self, queue_obj_id, lock_ids):
+        self.queues.add_item(lock_ids, DummyObject(queue_obj_id))
+        self._check_queue_dicts(queue_obj_id, lock_ids, -1)
+
+    def _check_queue_dicts(self, queue_obj_id, lock_ids, location=None, present=True):
+        """
+        if location is None, it looks anywhere. If it is an int, it'll look in that spot
+        present determines whether it's expected to be in the queue_by_lock_id or not
```

Comment:
`present` is ignored if `bool(location) == True`. Does that matter?

Reply:
Nope, not for my use case. Since it's easy enough, I might as well make it matter though